lib/routes/account.js

```diff
@@ -1911,8 +1982,199 @@ module.exports = function (
+        },
+        validate: {
+          payload: {
+            email: validators.email().required()
+          }
+        },
+        response: {}
+      },
+      handler: function (request, reply) {
+        const sessionToken = request.auth.credentials
+        const uid = sessionToken.uid
+        const primaryEmail = sessionToken.email
+        const ip = request.app.clientAddress
+        const email = request.payload.email
+        let emailCode, emailData
+
+        if (! sessionToken.emailVerified) {
+          return reply(error.unverifiedAccount())
+        }
+
+        if (sessionToken.tokenVerificationId) {
+          return reply(error.unverifiedSession())
+        }
+
+        if (sessionToken.email.toLowerCase() === email.toLowerCase()) {
+          return reply(error.yourPrimaryEmailExists())
+        }
+
+        log.begin('Account.RecoveryEmailCreate', request)
+
+        customs.check(request, primaryEmail, 'createEmail')
+          .then(checkEmail)
+          .then(generateRandomValues)
+          .then(createEmail)
+          .then(sendEmailVerification)
+          .then(
+            function () {
+              reply({})
+            },
             reply
           )
+
+        function checkEmail() {
+          return db.emailRecord(email)
+            .then((emailRecord) => {
+              if (emailRecord.emailVerified) {
+                throw error.verifiedPrimaryEmailAlreadyExists()
+              }
+
+              // Check to see if this account has been unverified for longer than
+              // a day, if so, delete account so another user can add it as a
+              // secondary email
+              const MS_IN_DAY = 1000*60*60*24
+              const msSinceCreated = Date.now() - emailRecord.createdAt
+              if (msSinceCreated >= MS_IN_DAY) {
+                return db.deleteAccount(emailRecord.uid)
+              } else {
+                throw error.unverifiedPrimaryEmailNewlyCreated()
+              }
+            })
+            .catch((err) => {
+              // Email does not exist in primary account table, carry on
+              if (err.errno !== error.ERRNO.ACCOUNT_UNKNOWN) {
+                throw err
+              }
+            })
+        }
+
+        function generateRandomValues() {
+          return random(16)
+            .then(bytes => {
+              emailCode = bytes
+            })
+        }
+
+        function createEmail() {
+          emailData = {
+            email: email,
+            normalizedEmail: email.toLowerCase(),
+            emailCode: emailCode,
+            isVerified: false,
+            isPrimary: false,
+            uid: uid
+          }
+          return db.createEmail(uid, emailData)
+        }
+
+        function sendEmailVerification() {
+          return getGeoData(ip)
+            .then((geoData) => {
+              return mailer.sendVerifySecondaryEmail([emailData], sessionToken, userAgent.call({
```

Comment:
You don't need to call `userAgent` here, you can re-use what's already been set on `sessionToken`.